### PR TITLE
Feat: Chat Form component

### PR DESCRIPTION
**Pull request by alamorre**

### CODE REVIEW

**Comment by alamorre** [comment]
I changed quite a bit. I think the jest files need a `.test.tsx` file patter to be picked up for tsdx testing. So I just made the files `.test.tsx` you don't event need something before the period.

File: src/ChatForm/styles.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    position: 'relative',
+    top: '4px',
+  } as Properties,
+  // closeIcon: {
+  //   position: 'relative',
+  //   top: '26px',
+  //   left: '18px',
+  //   fontSize: '16px',
+  //   fontFamily: 'Avenir',
+  // } as Properties,
```
Comment (dhavelock) [inline src/ChatForm/styles.tsx:21]: commented code

File: tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
-  "include": ["src", "types"],
+  "include": ["src", "types", "src/Button/.test.tsx", "src/ChatCard/.test.tsx", "src/Input/.test.tsx", "src/ChatForm/.test.tsx"],
```
Comment (dhavelock) [inline tsconfig.json:3]: is there some regex we can use here instead?

**Review by dhavelock — APPROVED** [review]

File: src/ChatForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  onFormSubmit,
+}) => {
+  const [selected, setSelected] = useState<boolean>(false);
+  const [value, setValue] = useState<string>('');
+
+  function onSubmit(event: React.SyntheticEvent) {
+    event.preventDefault();
+    onFormSubmit && onFormSubmit(value);
+    setValue('');
+  }
```
[the remark this answers is not in the document]
Comment (alamorre) [reply src/ChatForm/index.tsx:23]: Good point, I can make a habit of doing const based functions everywhere and I can clean this up within another merge 👍

File: src/ChatForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  return (
+    <div
+      className="ce-chat-form"
+      style={{
+        ...styles.chatForm,
+        ...style,
+        // Not sure why this is here, might need it later
+        // ...{ marginLeft: props.onClose ? '40px' : '0px' },
```
[the remark this answers is not in the document]
Comment (alamorre) [reply src/ChatForm/index.tsx:36]: Yeah I might need that for later, it's in `react-chat-engine` I just forget why haha

I'll take it out if it loves for like a month unchanged.

File: tsconfig.json (hunk 2 above)
Comment (alamorre) [reply tsconfig.json:3]: I didn't realize that got added. I'm gonna try and take it out - shouldn't be there imo.

File: src/ChatForm/.test.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (alamorre) [reply src/ChatForm/.test.tsx:1]: Yep thats gonna be here for now ^^ I added a comment explaining it

**Comment by dhavelock** [comment]
> I changed quite a bit. I think the jest files need a `.test.tsx` file patter to be picked up for tsdx testing. So I just made the files `.test.tsx` you don't event need something before the period.

ohhh I see- there is almost certainly a config we could specify for test file patterns for tsdx. Maybe in the future if we want to change but let's leave it for now 🚀

**Pull request merged**